.gitlab/functional_test/regression_detector.yml

```diff
@@ -128,7 +128,7 @@ single-machine-performance-regression_detector:
     # invoke task has additional logic that does not seem to apply well to SMP's
     # JUnit XML. Agent CI seems to use `datadog-agent` as the service name when
     # uploading JUnit XML, so the upload command below respects that convention.
-    - DATADOG_API_KEY="$("$CI_PROJECT_DIR"/tools/ci/fetch_secret.sh "$API_KEY_ORG2")" || exit $?; export DATADOG_API_KEY
+    - DATADOG_API_KEY="$("$CI_PROJECT_DIR"/tools/ci/fetch_secret.sh "$AGENT_API_KEY_ORG2" token)" || exit $?; export DATADOG_API_KEY
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I see we sometimes wrap the values with `"`, but not everywhere. Could we have some values with a space that would be break the script if we don't?

Reply:
No we don't have values with space. I think in all situations it's always better to wrap with `"`, I'll do a dedicated PR to make it more uniform (I think it would be nice if we could run the `shellcheck` on the `script` part of gitlab configuration

Reply:
@FlorentClarret I've been doing much of the maintenance on this particular CI job definition. You're right; the use of double quotes to wrap some environment variables has been inconsistent. Some of this inconsistency is an artifact of copying and pasting workflows from other CI systems (e.g., SMP's own repo) that have accumulated changes from multiple authors and don't consistently use quoting. I agree that using `shellcheck` would be worthwhile, and that a consistent quoting convention would be preferable. I also agree with @chouetz that a dedicated PR is best for establishing a quoting convention, and some automation to ensure that quoting convention is enforced in CI would be helpful, if it can be done with a practical amount of effort. Such automation would help ensure that future changes also comply with that convention.